public/app/features/dashboard/api/dashboard_api.ts

```diff
@@ -41,11 +43,21 @@ class LegacyDashboardAPI implements DashboardAPI {
     return getBackendSrv().delete<DeleteDashboardResponse>(`/api/dashboards/uid/${uid}`, { showSuccessAlert });
   }
 
-  getDashboardDTO(uid: string): Promise<DashboardDTO> {
+  async getDashboardDTO(uid: string): Promise<DashboardDTO> {
     const scopes = getSelectedScopesNames();
     const queryParams = scopes.length > 0 ? { scopes } : undefined;
 
-    return getBackendSrv().get<DashboardDTO>(`/api/dashboards/uid/${uid}`, queryParams);
+    const response = await lastValueFrom(
+      getBackendSrv().fetch<DashboardDTO>({
+        url: `/api/dashboards/uid/${uid}`,
+        params: queryParams,
+      })
+    );
+
+    return {
+      ...response.data,
+      reloadOnScopesChange: response.headers.has('X-Reload-Scopes-Change'),
```

Comment:
if not a dashboard schema property it could be meta property (returned by API )

Reply:
We discussed about adding a prop to the dashboard JSON but given this is more of an edge case rather than an actual option people would use, we decided on doing it via HTTP headers. I don't have a strong preference here.

What do you think @bergquist ?

Reply:
Moved away from HTTP headers approach in favor of a prop in `meta`